packages/account-v2/src/modules/src/IDVService/IDVService.tsx

```diff
@@ -1,65 +1,89 @@
 import React from 'react';
 import { Form, Formik, FormikHelpers } from 'formik';
+import { twMerge } from 'tailwind-merge';
 import { InferType } from 'yup';
-import { useIdentityDocumentVerificationAdd, useSettings } from '@deriv/api-v2';
+import { useIdentityDocumentVerificationAdd, useKycAuthStatus, useSettings } from '@deriv/api-v2';
 import { TSocketError } from '@deriv/api-v2/types';
 import { Button, Divider, InlineMessage, Loader, Text } from '@deriv-com/ui';
+import { POI_SERVICE } from '../../../constants';
 import { PersonalDetailsFormWithExample } from '../../../containers';
-import { TSupportedDocuments } from '../../../types';
 import {
+    checkIDVErrorStatus,
     generateIDVPayloadData,
     generateNameDOBFormData,
     generateNameDOBPayloadData,
+    getButtonText,
     getIDVFormValidationSchema,
     getNameDOBValidationSchema,
     setErrorMessage,
+    translateErrorCode,
 } from '../../../utils';
 import { IDVForm } from '../IDVForm';
 
 type TIDVServiceProps = {
-    countryCode: string;
+    countryCode?: string;
     handleComplete: () => void;
     onCancel: () => void;
-    supportedDocuments: TSupportedDocuments;
 };
 
 type TErrorData = TSocketError<'identity_verification_document_add' | 'set_settings'>;
 
-export const IDVService = ({ countryCode, handleComplete, onCancel, supportedDocuments }: TIDVServiceProps) => {
+export const IDVService = ({ countryCode, handleComplete, onCancel }: TIDVServiceProps) => {
     const {
         data: personalInfo,
         isLoading,
         mutation: { mutateAsync: updateAsync },
     } = useSettings();
     const { mutateAsync: submitIDVDocumentsAsync } = useIdentityDocumentVerificationAdd();
 
-    const idvValidationSchema = getIDVFormValidationSchema(countryCode, supportedDocuments, {});
-    const personalDetailsValidationSchema = getNameDOBValidationSchema();
-    const idvServiceSchema = personalDetailsValidationSchema.concat(idvValidationSchema);
+    const payload = countryCode ? { country: countryCode } : undefined;
```

Comment:
Hi @likhith-deriv 👋 
If we pass undefined as a payload will it give meaningful response ?

Reply:
For the same hook, in some scenarios we dont want the payload to be sent. Hence I'm making a check and sending a payload or undefined